cmd/erasure-createfile.go

```diff
@@ -80,7 +80,7 @@ func (s *ErasureStorage) CreateFile(src io.Reader, volume, path string, buffer [
 
 // erasureAppendFile appends the content of buf to the file on the given disk and updates computes
 // the hash of the written data. It sends the write error (or nil) over the error channel.
-func erasureAppendFile(disk StorageAPI, volume, path string, hash hash.Hash, buf []byte, errChan chan<- error) {
+func erasureAppendFile(disk StorageAPI, volume, path string, hashWriter io.Writer, buf []byte, errChan chan<- error) {
```

Comment:
In this case I'm not sure because the semantic of erasureAppendFile requires a hash function (actually not necessarily but minio only supports hash functions for bitrot protection). The variable name / doc gives some hint but a type would be more explicit...
The question here is: explicitness vs. abstraction...